src/charm.py

```diff
@@ -314,6 +322,106 @@ def _handle_non_online_instance_status(self, state) -> None:
             except SnapServiceOperationError as e:
                 self.unit.status = BlockedStatus(e.message)
 
+    def _handle_potential_change_in_ip_address(self) -> bool:
+        """Helper method to handle potential change in unit ip address.
+
+        Returns a boolean indicating whether the report_host is the same as the
+        unit ip address.
+        """
+        if not self._mysql.is_mysqld_running():
+            # return true to allow other self-healing scenarios in update-status
+            # to proceed
+            return True
+
+        try:
+            report_host = self._mysql.get_variable_value("report_host")
+        except MySQLGetVariableValueError:
+            report_host = None
+
+        try:
+            hostname = socket.gethostname()
+            unit_ip = socket.gethostbyname(hostname)
```

Comment:
I think it is possible to get `127.0.0.1` here under some `/etc/hosts` configuration.
maybe https://stackoverflow.com/a/28950776